cli/compose/types/types.go

```diff
@@ -62,6 +65,74 @@ type ConfigDetails struct {
 	Environment map[string]string
 }
 
+// Duration is a thin wrapper around time.Duration with improved JSON marshalling
+type Duration struct {
```

Comment:
`type Duration time.Duration`

Reply:
Also, should be declared below `Config`, … (even better would be to split this file but it can be in a follow-up

Reply:
I've tried this, but `type Duration time.Duration` doesn't let me inherit the `String()` method, which I need to properly marshal the object.

Reply:
I think this would work:

```go
time.Duration(d).String()
```

Reply:
As @dnephin mentionned, this works

```
package main

import (
	"fmt"
	"time"
)

type Duration time.Duration

func (d Duration) String() string {
	return "foo:" + time.Duration(d).String()
}

func main() {
	t := time.Duration(2000)
	fmt.Println("Hello, playground")
	fmt.Println(t)
	fmt.Println(Duration(t))
}
```

Reply:
So in this case 

```go

// Duration is a thin wrapper around time.Duration with improved JSON marshalling
type Duration time.Duration

// MarshalJSON makes Duration implement json.Marshaler
func (d Duration) MarshalJSON() ([]byte, error) {
	return json.Marshal(d.String())
}

// MarshalYAML makes Duration implement yaml.Marshaler
func (d Duration) MarshalYAML() (interface{}, error) {
	return d.String(), nil
}

func (d Duration) String() string {
	return time.Duration(d).String()
}
```

… and thus no need for the `ConvertDuration` anymor

Reply:
@shin- *wrapping* works just fine.

```go	
td := time.Duration(10)
cd := Duration(10)
fmt.Println(td)
fmt.Println(cd)
fmt.Println(Duration(td))
fmt.Println(time.Duration(td))
```

Reply:
Yep, but we have several structs that use pointers, and those need to be converted. See https://github.com/docker/cli/pull/1326/files#diff-b3114871e359bbcd13674fefe7b3e16eR288 for example.